homeassistant/components/remote/xiaomi_miio.py

```diff
@@ -0,0 +1,257 @@
+            title='Xiaomi Miio Remote')
+
+    hass.services.async_register(DOMAIN, SERVICE_LEARN, async_service_handler,
+                                 schema=LEARN_COMMAND_SCHEMA)
+
+
+class XiaomiMiioRemote(RemoteDevice):
+    """Representation of a Xiaomi Miio Remote device."""
+
+    def __init__(self, friendly_name, device,
+                 slot, timeout, hidden, commands):
+        """Initialize the remote."""
+        self._name = friendly_name
+        self._device = device
+        self._is_hidden = hidden
+        self._slot = slot
+        self._timeout = timeout
+        self._state = False
+        self._commands = commands
+
+    @property
+    def name(self):
+        """Return the name of the remote."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the remote object."""
+        return self._device
+
+    @property
+    def hidden(self):
+        """Return if we should hide entity."""
+        return self._is_hidden
+
+    @property
+    def slot(self):
+        """Return the slot to save learned command."""
+        return self._slot
+
+    @property
+    def timeout(self):
+        """Return the timeout for learning command."""
+        return self._timeout
+
+    @property
+    def is_on(self):
+        """Return False if device is unreachable, else True."""
+        from miio import DeviceException
+        try:
+            self.device.info()
+            return True
+        except DeviceException:
+            return False
+
+    @property
+    def should_poll(self):
+        """We should not be polled for device up state."""
+        return False
+
+    @property
+    def device_state_attributes(self):
+        """Hide remote by default."""
+        if self._is_hidden:
+            return {'hidden': 'true'}
+        else:
+            return
+
+    # pylint: disable=R0201
+    @asyncio.coroutine
+    def async_turn_on(self, **kwargs):
+        """Turn the device on."""
+        _LOGGER.error("Device does not support turn_on, " +
+                      "please use 'remote.send_command' to send commands.")
+
+    @asyncio.coroutine
+    def async_turn_off(self, **kwargs):
+        """Turn the device off."""
+        _LOGGER.error("Device does not support turn_off, " +
+                      "please use 'remote.send_command' to send commands.")
+
+    # pylint: enable=R0201
+    def _send_command(self, payload):
+        """Send a command."""
+        from miio import DeviceException
+
+        _LOGGER.debug("Sending payload: '%s'", payload)
+        try:
+            self.device.play(payload)
+            return True
+        except DeviceException as ex:
+            _LOGGER.error(
+                "Transmit of IR command failed, %s, exception: %s",
+                payload, ex)
+            return False
+
+    def send_command(self, command, **kwargs):
+        """Wrapper for _send_command."""
+        num_repeats = kwargs.get(ATTR_NUM_REPEATS)
```

Comment:
Should this default to 1 ?

Reply:
It already does. https://github.com/home-assistant/home-assistant/blob/a3e36e6c66dff6eb33bc15847b7666df0b098eeb/homeassistant/components/remote/__init__.py#L46